src/Microsoft.ML.StandardLearners/Standard/ModelStatistics.cs

```diff
@@ -27,12 +27,12 @@ namespace Microsoft.ML.Learners
     public readonly struct CoefficientStatistics
     {
         public readonly string Name;
```

Comment:
>public readonly string Name; [](start = 8, length = 28)

As discussed, if this is to be part of our public API we should probably as part of this PR make this entire structure a bit less goofy. If I were to trace back the root of what makes this structure a confusing structure, I think this choice here -- having the name of the slot instead of just the slot index, is the original sin.

This should have just been `The fact that this is a `Name` instead of just a `SlotIndex` is the cause for an enormous amount of confusion in this structure. This requires that we link it with the schema column for features, but from an API perspective this is just something that could have easily (in fact, more easily) be done by the user themselves. As it is, the structure is offering "convenience" but in a way that inevitably leads to a more complicated API.

Once you do that, restructuring the odd API below (which I'll comment more on in the test file) should become more approachable.

Reply:
This is not to say that it might be convenient to have an internal utility method that makes saving as text more convenient for the `LinearModelParameters` and whatnot, but confusing the needs of the text exporter (which is not part of our public API) with what we actually do in the public API was a misstep.

---
In reply to: [244374085](https://github.com/dotnet/machinelearning/pull/1979#discussion_r244374085) [](ancestors = 244374085)